packages/toolkit/src/view/knowledge/components/CatalogFilesTab.tsx

```diff
@@ -0,0 +1,508 @@
+  });
+
+  const deleteKnowledgeBaseFile = useDeleteKnowledgeBaseFile();
+  const [showDeleteMessage, setShowDeleteMessage] = React.useState(false);
+  const [fileToDelete, setFileToDelete] = React.useState<File | null>(null);
+  const timeoutRef = React.useRef<NodeJS.Timeout | null>(null);
+  const [isFileDetailsOpen, setIsFileDetailsOpen] = React.useState(false);
+  const [selectedFile, setSelectedFile] = React.useState<File | null>(null);
+
+  React.useEffect(() => {
+    let interval: NodeJS.Timeout | null = null;
+
+    if (files && files.some(file => file.processStatus !== "FILE_PROCESS_STATUS_COMPLETED")) {
+      interval = setInterval(() => {
+        refetch();
+      }, 5000);
+    }
+
+    return () => {
+      if (interval) {
+        clearInterval(interval);
+      }
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+      }
+    };
+  }, [files, refetch]);
+
+  const handleFileClick = (file: File) => {
+    setSelectedFile(file);
+    setIsFileDetailsOpen(true);
+  };
+
+  const closeOverlay = () => {
+    setSelectedFile(null);
+    setIsFileDetailsOpen(false);
+  };
+
+  const getStatusSortValue = (status: FileStatus): number => {
+    const statusOrder = {
+      NOTSTARTED: 0,
+      WAITING: 1,
+      CONVERTING: 2,
+      CHUNKING: 3,
+      EMBEDDING: 4,
+      COMPLETED: 5,
+      FAILED: 6,
+    };
+    return statusOrder[status] ?? -1;
+  };
+
+  const sortedData = React.useMemo(() => {
+    if (!files) return [];
+    return [...files].sort((a, b) => {
+      if (sortConfig.key === "") return 0;
+
+      let aValue: any = a[sortConfig.key];
+      let bValue: any = b[sortConfig.key];
+
+      if (sortConfig.key === "processStatus") {
+        aValue = getStatusSortValue(a.processStatus.replace("FILE_PROCESS_STATUS_", "") as FileStatus);
+        bValue = getStatusSortValue(b.processStatus.replace("FILE_PROCESS_STATUS_", "") as FileStatus);
+      } else if (sortConfig.key === "size" || sortConfig.key === "totalChunks" || sortConfig.key === "totalTokens") {
+        aValue = Number(aValue);
+        bValue = Number(bValue);
+      } else if (sortConfig.key === "createTime") {
+        aValue = new Date(aValue).getTime();
+        bValue = new Date(bValue).getTime();
+      }
+
+      if (aValue < bValue) {
+        return sortConfig.direction === "ascending" ? -1 : 1;
+      }
+      if (aValue > bValue) {
+        return sortConfig.direction === "ascending" ? 1 : -1;
+      }
+      return 0;
+    });
+  }, [files, sortConfig]);
+
+  const handleDelete = (fileUid: string) => {
+    const file = files?.find((item) => item.fileUid === fileUid);
+    if (file) {
+      setFileToDelete(file);
+      setShowDeleteMessage(true);
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+      }
+      timeoutRef.current = setTimeout(() => {
+        setShowDeleteMessage(false);
+        actuallyDeleteFile(file);
+      }, DELETE_FILE_TIMEOUT);
+    }
+  };
```

Comment:
@thewbuk There are multiple place in your implementation that use this kind of pattern. I am a bit confused with several reasons

- We use toast notification instead of manual maintaining a notification state
- Why you need to show delete file notification then actually delete it 🤔 This is too optimistic. What if you didn't delete it?